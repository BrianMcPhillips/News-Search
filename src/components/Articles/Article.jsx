import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ 
  title, description, author, publishedAt, 
  url, imageUrl, content 
}) => (
  <>
    <h2>{title}</h2>
    <h3>{description}</h3>
    <h4>{author}</h4>
    <p>{publishedAt}</p>
    <p>{url}</p>
    <img src={imageUrl} alt={title} />
    <p>{content}</p>
  </>
);

Article.propTypes = {
  
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  publishedAt: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired
  
  
};

export default Article;
