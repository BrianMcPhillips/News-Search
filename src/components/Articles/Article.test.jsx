import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Article from './Article';

describe('Article component', () => {
  afterEach(() => cleanup());
  it('renders Article', () => {
    const { asFragment } = render(<Article 
      title="Test Article"
      description="This is a test description"
      author="Test Testerson"
      publishedAt="12/12/2020"
      url="testurl.com/test/test"
      imageUrl="http://placeholder.com/article"
      content="This is the best test article ever written"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
