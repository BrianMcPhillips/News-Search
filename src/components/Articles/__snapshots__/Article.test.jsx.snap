// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders Article 1`] = `
<DocumentFragment>
  <h2>
    Test Article
  </h2>
  <h3>
    This is a test description
  </h3>
  <h4>
    Test Testerson
  </h4>
  <p>
    12/12/2020
  </p>
  <p>
    testurl.com/test/test
  </p>
  <img
    alt="Test Article"
    src="http://placeholder.com/article"
  />
  <p>
    This is the best test article ever written
  </p>
</DocumentFragment>
`;
